src/io/flutter/editor/OutlineLocation.java

```diff
@@ -12,22 +12,50 @@
 import org.dartlang.analysis.server.protocol.FlutterOutline;
 import org.jetbrains.annotations.Nullable;
 
-/**
- * Class that tracks the location of a FlutterOutline node in a document.
- * <p>
- * Once the track method has been called, edits to the document are reflected
- * by by all locations returned by the outline location.
- */
-public class OutlineLocation implements Comparable<OutlineLocation> {
-  private final int line;
-  private final int column;
-  private final int indent;
-  private final int offset;
-  private final int endOffset;
-  @Nullable
+class TextRangeTracker {
```

Comment:
This helper class was extracted out as we now want to track two text ranges per outline location. One to render the ui guides and one that tracks the entire body of the outline node that is used for the inline widget previews.